### PR TITLE
ci: add artifact scanning, windows test and check in e2e support changes

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
This checks in the most recent e2e changes, and adds base support for windows.
Also added artifact scanning.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/scan_artifacts.ts
Comment (kuhlmanp) [inline .circleci/scan_artifacts.ts:1]: Not sure modifying this is a good idea. We probably need to keep this around as long as we're in CircleCI. Probably better to make a copy of this name for codebuild specifically.

**Review by kuhlmanp — APPROVED** [review]

File: .circleci/scan_artifacts_codebuild.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+// This list contains a platform agnostic list of paths where artifacts are stored after each test.
+import { ARTIFACT_STORAGE_PATH_ALLOW_LIST_CODEBUILD } from '../scripts/artifact-storage-path-allow-list-codebuild';
+
+const ROOT_FOLDER_ABSOLUTE_PATH = process.env.CODEBUILD_SRC_DIR as string;
+
+export const hasMatchingContentInFolder = (
+  patterns: string[],
+  folder: string,
+  excludeFolder = '{node_modules,.cache,.git,.cache,.git}',
```
Comment (evcodes) [inline .circleci/scan_artifacts_codebuild.ts:12]: why is .cache and .git repeated here?

Comment (evcodes) [inline .circleci/scan_artifacts_codebuild.ts:12]:
```suggestion
  excludeFolder = '{node_modules,.cache,.git}',
```

File: shared-scripts.sh [hunk 2]
```diff
@@ -298,15 +298,36 @@ function _runE2ETestsWin {
...
     source .circleci/local_publish_helpers.sh && startLocalRegistry "$CODEBUILD_SRC_DIR/.circleci/verdaccio.yaml"
     source $BASH_ENV
 
     setNpmRegistryUrlToLocal
     changeNpmGlobalPath
     amplify version
-    #should just fail here..
+
     cd packages/amplify-e2e-tests
+
```
Comment (evcodes) [inline shared-scripts.sh:320]:
```suggestion
```

File: shared-scripts.sh [hunk 3]
```diff
@@ -298,15 +298,36 @@ function _runE2ETestsWin {
...
+    _loadTestAccountCredentials
+
```
Comment (evcodes) [inline shared-scripts.sh:322]:
```suggestion
```

File: shared-scripts.sh [hunk 4]
```diff
@@ -284,7 +284,7 @@ function _runE2ETestsLinux {
 
     retry runE2eTest
 }
-function _runE2ETestsWin {
+function _runE2ETestsWindows {
     echo RUN E2E Tests Windows
```
Comment (evcodes) [inline shared-scripts.sh:289]:
```suggestion
```

File: shared-scripts.sh [hunk 5]
```diff
@@ -298,15 +298,36 @@ function _runE2ETestsWin {
...
     changeNpmGlobalPath
     amplify version
-    #should just fail here..
+
     cd packages/amplify-e2e-tests
+
+    _loadTestAccountCredentials
+
     retry runE2eTest
 }
```
Comment (evcodes) [inline shared-scripts.sh:325]:
```suggestion
```

**Review by evcodes — APPROVED** [review]
Minor nits but overall LGTM

File: .circleci/scan_artifacts_codebuild.ts (hunk 1 above)
Comment (awsluja) [reply .circleci/scan_artifacts_codebuild.ts]: fixed, thanks

**Pull request merged**